src/shady-css-parser/common.js

```diff
@@ -0,0 +1,36 @@
+let matcher = {
+  whitespace: /\s/,
+  commentOpen: /^\/\*/,
+  commentClose: /\*\/$/,
+  boundary: /[\(\)\{\}\[\]'"@;:\s\.#+>~\^]|\/\*/,
+  blockBoundary: /[;\{\}]/,
+  propertyBoundary: /^[;:]/,
+  propertyDelimiter: /^[;\}]/,
+  stringBoundary: /['"]/
+};
+
+let tokenType = {
+  none: 0,
+  whitespace: 1,
+  boundary: 2,
+  stringBoundary: 3,
+  string: 4,
+  word: 5
+};
+
+let whitespaceGreedy = /(\s+)/g;
+
+function traverseWhitespace(source, offset) {
+  var match;
+
+  whitespaceGreedy.lastIndex = offset;
```

Comment:
This is old and should be removed. The function, not JS RegExp ;) although the same applies.